File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Pages/LanguageView.swift
import SwiftUI

struct LanguageView: View {
    @Environment(\.presentationMode) var presentationMode

    @State private var selectedLanguage = "en"

    // Brand Colors
    private let primaryBlue = Color(red: 37 / 255, green: 99 / 255, blue: 235 / 255)

    let languages = [
        Language(code: "en", name: "English", nativeName: "English", subtext: "Default"),
        Language(code: "hi", name: "Hindi", nativeName: "हिंदी", subtext: "हिंदी में देखें"),
    ]

    var body: some View {
        VStack(spacing: 0) {
            // Header
            HStack {
                Button(action: {
                    presentationMode.wrappedValue.dismiss()
                }) {
                    Image(systemName: "arrow.left")
                        .font(.system(size: 20, weight: .medium))
                        .foregroundColor(Color(hex: "#111827"))
                }

                Spacer()

                Text("Select Language")
                    .font(.system(size: 18, weight: .bold))
                    .foregroundColor(Color(hex: "#111827"))

                Spacer()

                Color.clear.frame(width: 20, height: 20)
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 14)
            .background(Color.white)
            .overlay(
                Rectangle()
                    .fill(Color(hex: "#E5E7EB"))
                    .frame(height: 1),
                alignment: .bottom
            )

            ScrollView {
                VStack(spacing: 16) {
                    Text("Choose your preferred language")
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(Color(hex: "#111827"))
                        .frame(maxWidth: .infinity, alignment: .leading)
                        .padding(.top, 24)

                    Text("You can change this anytime from settings")
                        .font(.system(size: 14))
                        .foregroundColor(Color(hex: "#6B7280"))
                        .frame(maxWidth: .infinity, alignment: .leading)
                        .padding(.bottom, 16)

                    ForEach(languages, id: \.code) { lang in
                        LanguageCard(language: lang, isSelected: selectedLanguage == lang.code) {
                            selectedLanguage = lang.code
                        }
                    }
                }
                .padding(16)
            }

            // Footer
            VStack {
                Button(action: {
                    // Save logic would go here
                    presentationMode.wrappedValue.dismiss()
                }) {
                    Text("Continue")
                        .font(.system(size: 16, weight: .semibold))
                        .foregroundColor(.white)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 16)
                        .background(primaryBlue)
                        .cornerRadius(30)
                }
            }
            .padding(20)
            .background(Color.white)
            .overlay(
                Rectangle()
                    .fill(Color(hex: "#E5E7EB"))
                    .frame(height: 1),
                alignment: .top
            )
        }
        .background(Color.white)
        .navigationBarHidden(true)
    }
}

struct Language {
    let code: String
    let name: String
    let nativeName: String
    let subtext: String
}

struct LanguageCard: View {
    let language: Language
    let isSelected: Bool
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack {
                VStack(alignment: .leading, spacing: 4) {
                    Text(language.nativeName)
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(isSelected ? Color(hex: "#1E40AF") : Color(hex: "#111827"))

                    Text(language.name)
                        .font(.system(size: 14))
                        .foregroundColor(isSelected ? Color(hex: "#3B82F6") : Color(hex: "#4B5563"))

                    Text(language.subtext)
                        .font(.system(size: 12, weight: .light))
                        .foregroundColor(isSelected ? Color(hex: "#3B82F6") : Color(hex: "#6B7280"))
                        .italic()
                }

                Spacer()

                ZStack {
                    Circle()
                        .stroke(
                            isSelected ? Color(hex: "#2563EB") : Color(hex: "#D1D5DB"), lineWidth: 2
                        )
                        .frame(width: 24, height: 24)

                    if isSelected {
                        Circle()
                            .fill(Color(hex: "#2563EB"))
                            .frame(width: 12, height: 12)
                    }
                }
            }
            .padding(20)
            .background(isSelected ? Color(hex: "#EFF6FF") : Color(hex: "#F9FAFB"))
            .cornerRadius(12)
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .stroke(isSelected ? Color(hex: "#2563EB") : Color.clear, lineWidth: 2)
            )
        }
        .buttonStyle(PlainButtonStyle())
    }
}

File: apps/Applications/android/app/src/main/java/com/ecommerceearn/app/ui/pages/LanguageView.kt
package com.ecommerceearn.app.ui.pages

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.ecommerceearn.app.data.manager.NavigationManager

data class Language(val code: String, val name: String, val nativeName: String, val subtext: String)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun LanguageView() {
    var selectedLanguage by remember { mutableStateOf("en") }

    val languages = listOf(
        Language("en", "English", "English", "Default"),
        Language("hi", "Hindi", "हिंदी", "हिंदी में देखें")
    )

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Select Language", fontSize = 18.sp, fontWeight = FontWeight.Bold, color = Color(0xFF111827)) },
                navigationIcon = {
                    IconButton(onClick = { NavigationManager.goBack() }) {
                        Icon(Icons.Default.ArrowBack, contentDescription = "Back", tint = Color(0xFF111827))
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = Color.White)
            )
        },
        bottomBar = {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(Color.White)
                    .padding(20.dp)
            ) {
                Divider() // acts as the top border from the iOS layout
                Spacer(modifier = Modifier.height(16.dp))
                Button(
                    onClick = { NavigationManager.goBack() },
                    modifier = Modifier.fillMaxWidth().height(50.dp),
                    shape = RoundedCornerShape(30.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF2563EB))
                ) {
                    Text("Continue", color = Color.White, fontSize = 16.sp, fontWeight = FontWeight.SemiBold)
                }
            }
        },
        containerColor = Color.White
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .verticalScroll(rememberScrollState())
                .padding(16.dp)
        ) {
            Text("Choose your preferred language", fontSize = 18.sp, fontWeight = FontWeight.Bold, color = Color(0xFF111827))
            Spacer(modifier = Modifier.height(4.dp))
            Text("You can change this anytime from settings", fontSize = 14.sp, color = Color(0xFF6B7280))
            Spacer(modifier = Modifier.height(16.dp))

            languages.forEach { lang ->
                LanguageCard(language = lang, isSelected = selectedLanguage == lang.code) {
                    selectedLanguage = lang.code
                }
                Spacer(modifier = Modifier.height(12.dp))
            }
        }
    }
}

@Composable
fun LanguageCard(language: Language, isSelected: Boolean, action: () -> Unit) {
    val bgColor = if (isSelected) Color(0xFFEFF6FF) else Color(0xFFF9FAFB)
    val borderColor = if (isSelected) Color(0xFF2563EB) else Color.Transparent

    Card(
        modifier = Modifier
            .fillMaxWidth()
            .clickable { action() },
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(containerColor = bgColor),
        border = BorderStroke(2.dp, borderColor)
    ) {
        Row(
            modifier = Modifier.padding(20.dp).fillMaxWidth(),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = language.nativeName,
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    color = if (isSelected) Color(0xFF1E40AF) else Color(0xFF111827)
                )
                Text(
                    text = language.name,
                    fontSize = 14.sp,
                    color = if (isSelected) Color(0xFF3B82F6) else Color(0xFF4B5563)
                )
                Text(
                    text = language.subtext,
                    fontSize = 12.sp,
                    fontWeight = FontWeight.Light,
                    fontStyle = FontStyle.Italic,
                    color = if (isSelected) Color(0xFF3B82F6) else Color(0xFF6B7280)
                )
            }

            Box(
                modifier = Modifier
                    .size(24.dp)
                    .clip(CircleShape)
                    .background(Color.Transparent)
                    .border(2.dp, if (isSelected) Color(0xFF2563EB) else Color(0xFFD1D5DB), CircleShape),
                contentAlignment = Alignment.Center
            ) {
                if (isSelected) {
                    Box(modifier = Modifier.size(12.dp).clip(CircleShape).background(Color(0xFF2563EB)))
                }
            }
        }
    }
}
